awx/main/tasks.py

```diff
@@ -771,23 +771,27 @@ def build_private_data_files(self, instance, **kwargs):
                 # For credentials used with ssh-add, write to a named pipe which
                 # will be read then closed, instead of leaving the SSH key on disk.
                 if credential and credential.kind in ('ssh', 'scm') and not ssh_too_old:
-                    name = 'credential_%d' % credential.pk
-                    path = os.path.join(kwargs['private_data_dir'], name)
-                    run.open_fifo_write(path, data)
+                    try:
```

Comment:
What's this about?

Reply:
bit of a safety net. Making the private_data_dir needs to happen early because everything gets dumped in there. Often times a feature is added that needs to do things earlier than other existing things. People often duplicate the creating of the private_data_dir. This protects against the private_data_dir already existing in that common scenario.